client-shaded/pom.xml

```diff
@@ -0,0 +1,143 @@
+                                    <shadedPattern>${shade.relocation.package}.google</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>io.grpc</pattern>
+                                    <shadedPattern>${shade.relocation.package}.io.grpc</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>io.perfmark</pattern>
+                                    <shadedPattern>${shade.relocation.package}.io.perfmark</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>javax.annotation</pattern>
+                                    <shadedPattern>${shade.relocation.package}.javax.annotation</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>net.openhft</pattern>
+                                    <shadedPattern>${shade.relocation.package}.net.openhft</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.codehaus</pattern>
+                                    <shadedPattern>${shade.relocation.package}.org.codehaus</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>okio</pattern>
+                                    <shadedPattern>${shade.relocation.package}.okio</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.checkerframework</pattern>
+                                    <shadedPattern>${shade.relocation.package}.org.checkerframework</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>org.reactivestreams</pattern>
+                                    <shadedPattern>${shade.relocation.package}.org.reactivestreams</shadedPattern>
+                                </relocation>
+                                <relocation>
+                                    <pattern>reactor</pattern>
+                                    <shadedPattern>${shade.relocation.package}.reactor</shadedPattern>
+                                </relocation>
+                            </relocations>
+                            <filters>
+                                <filter>
+                                    <artifact>*:*</artifact>
+                                    <excludes>
+                                        <exclude>module-info.class</exclude>
+                                        <exclude>META-INF/MANIFEST.MF</exclude>
```

Comment:
The resulting shaded jar contains some unusable files that could be excluded.
checked with these commands:
```
mvn clean install -DskipTests
unzip -Z -C -1 client-shaded/target/client-shaded-0.1.2-SNAPSHOT.jar|grep -v io/streamnative
```
output of this command: https://gist.github.com/lhotari/4e051b060c4aa976d132423285fc0384

possible exclusions:
META-INF/native/
META-INF/native-image/
META-INF/proguard/
META-INF/maven/

Reply:
The included netty native libraries are part of gprc-netty which is already shaded and that gets shaded again.

Reply:
> The included netty native libraries are part of gprc-netty which is already shaded and that gets shaded again. 

It's not ideal, though if we don't do it, there could be conflicts with different grpc/netty versions

Reply:
I guess the native libraries should be relocated too?

Reply:
We still need to have this PR otherwise the pom for non-shaded version is broken. 

I've removed grpc-netty-shaded from shaded artifact